expression/builtin_miscellaneous_vec_test.go

```diff
@@ -33,7 +33,9 @@ var vecBuiltinMiscellaneousCases = map[string][]vecExprBenchCase{
 	ast.Inet6Ntoa:    {},
 	ast.InetAton:     {},
 	ast.IsIPv4Mapped: {},
-	ast.IsIPv4Compat: {},
+	ast.IsIPv4Compat: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString}},
+	},
```

Comment:
The datagen will generate random string, and almost all of them will return `0` because it is invalid. We need more test cases to judge the valid ip string. Use `ipv6StrGener` and implement a new `ipv4StrGener`?

Reply:
Thanks for the guidence. I am looking into it. Just got a little confused where should I put the expect value of the test case.

Reply:
cc L25-27 , `Inet6Aton` use `ipv6StrGener` to generate data.  Of course, you can look other vec_test.go file, and there are many examples.

Reply:
Thanks, it really helps. I implement `ipv4CompatByteGener` to return IPv4 compatible IPv6 address. I also implemnt `ipv4ByteGener` and `ipv6ByteGener`. The reason that I cannot reuse the `ipv6StrGener` is `net.IP.String()` method will insert `.` or `::` into the result string, which should be part of the input of `IsIPv4Compat` function